fmf/_compat_jsonschema.py

```diff
@@ -0,0 +1,91 @@
+"""Compatibility layer for jsonschema validation."""
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, Dict, List, NamedTuple, Optional, Union
+
+import jsonschema
+
+try:
+    from jsonschema import validators
+    from referencing import Registry, create
+    from referencing.jsonschema import DRAFT4
+    MODERN_JSONSCHEMA = True
+except ImportError:
+    MODERN_JSONSCHEMA = False
+
+
+class JsonSchemaValidationResult(NamedTuple):
+    """Represents JSON Schema validation result."""
+    result: bool
+    errors: List[Any]
+
+
+def create_validator(
+        schema: Dict[str, Any],
+        schema_store: Optional[Dict[str, Any]] = None
+        ) -> Union[jsonschema.validators.Validator, jsonschema.validators.Draft4Validator]:
+    """Create a validator instance based on available jsonschema version."""
+    schema_store = schema_store or {}
+
+    if MODERN_JSONSCHEMA:
+        # Modern approach with referencing
+        registry: Registry = Registry().with_resources([
```

Comment:
```suggestion
        registry = Registry().with_resources([
```

Should not be necessary